python/pyspark/context.py

```diff
@@ -244,26 +269,32 @@ def _do_init(
         self.environment["PYTHONHASHSEED"] = os.environ.get("PYTHONHASHSEED", "0")
 
         # Create the Java SparkContext through Py4J
-        self._jsc = jsc or self._initialize_context(self._conf._jconf)
+        self._jsc = jsc or self._initialize_context(self._conf._jconf)  # type: ignore[attr-defined]
         # Reset the SparkConf to the one actually used by the SparkContext in JVM.
         self._conf = SparkConf(_jconf=self._jsc.sc().conf())
 
         # Create a single Accumulator in Java that we'll send all our updates through;
         # they will be passed back to us through a TCP server
-        auth_token = self._gateway.gateway_parameters.auth_token
-        self._accumulatorServer = accumulators._start_update_server(auth_token)
+        auth_token = cast(JVMView, self._gateway).gateway_parameters.auth_token
+        self._accumulatorServer = accumulators._start_update_server(  # type: ignore[attr-defined]
+            auth_token
+        )  # type: ignore[attr-defined]
         (host, port) = self._accumulatorServer.server_address
-        self._javaAccumulator = self._jvm.PythonAccumulatorV2(host, port, auth_token)
+        self._javaAccumulator = cast(JVMView, self._jvm).PythonAccumulatorV2(host, port, auth_token)
```

Comment:
Given how many of these casts we have in this scope, it might be better to extract a local variable

```python
jvm = cast(JVMView, self._jvm)
```